### DISCUSSION TITLE
How do I swap apostrophe and semicolon?

### DISCUSSION THREAD

**Discussion by mr-scrpt** [manna-harbour/miryoku#182] [Q&A] [answered]
I am using a Corne 5 x 3 + 3 thumb buttons. To create the firmware I use the "Build Inputs" action, in it I select the parameters for my keyboard for example:
![image](https://user-images.githubusercontent.com/48452376/209646698-c1f42474-175e-411e-ab88-cbadc4b778b0.png)
But I default apostrophe and semicolon swapped - I'd like to avoid this swap because on a Cyrillic keyboard the **;** symbol is on the letter, the basic layout is very awkward in this case

I'd really like to be able to change this via action parameters, but if that's not possible. Can you tell me how this can be changed manually?

**Comment by manna-harbour** [comment, marked as answer]
There's no built-in option yet. See https://github.com/manna-harbour/miryoku/issues/35.

For now you could substitute custom Base, Num, and Sym layers. See  https://github.com/manna-harbour/miryoku/discussions/85.

Also, for the build shown above, note that you can build for left and right in the same run with `corne_left,corne_right`, if you're only updating the keymap you only need to build and flash left, and the Extra layer is QWERTY by default (see https://github.com/manna-harbour/miryoku/tree/master/docs/reference#additional-features).

**Comment by RascasseTabac** [comment]
If you compile the QMK firmware yourself, you can do this using key overrides in keymap.c:
const key_override_t colon_key_override =
    ko_make_basic(MOD_MASK_SHIFT, KC_COLON, KC_SEMICOLON);
 const key_override_t** key_overrides = (const key_override_t*[]){
    &colon_key_override,
    NULL
}; 
Then use KC_COLON in your keymaps.

**Reply by Pe8er** [reply]
Is it possible in ZMK?
